src/ResourceGenerator/ExtractCollection.php

```diff
@@ -76,6 +74,15 @@ private function extractPaginator(
                 ? ($request->getQueryParams()[$paginationParam] ?? 1)
                 : $request->getAttribute($paginationParam, 1);
 
+            if ($page < 1 || ($page > $pageCount && $pageCount > 0)) {
+                throw new OutOfBoundsException(sprintf(
```

Comment:
I think we should add `@throws` tag with that exception in PHPDoc function comment.

What more, maybe we should use also here 'library specific exception', so we need to create new exception, which extends `OutOfBoundsException` and implements `Zend\Expressive\Hal\Exception\Exception`?

Reply:
Agreed! I will commit the changes soon. Thanks.